### PR TITLE
V3 string functions

### PR DESCRIPTION
**Pull request by naterush**
# Description

Adds v3 of string functions. 

# Testing

I tried to test every combo, but I am sure I forgot some. Maybe try and add 2 tests per?

# Documentation

N/a.

### CODE REVIEW

File: mitosheet/mitosheet/public/v3/sheet_functions/string_functions.py
Comment (naterush) [inline mitosheet/mitosheet/public/v3/sheet_functions/string_functions.py:1]: Very interestingly, we only gained about 20 lines of code. But we add so much more functionality that works so much better - I'm pretty proud of this refactor.

File: mitosheet/mitosheet/public/v3/sheet_functions/string_functions.py [hunk 1]
```diff
@@ -73,46 +189,398 @@ def LEFT(string: StringRestrictedInputType, num_chars: IntRestrictedInputType=No
...
-        lambda previous_value, new_value: previous_value + new_value,
-        lambda previous_series, new_series: previous_series + new_series
-    )
+    if string is None:
+        return ''
+    elif isinstance(string, pd.Series):
+        string = string.fillna('')
+
+    if start_loc is None:
+        start_loc = 0
```
[the remark this answers is not in the document]
Comment (aarondr77) [reply mitosheet/mitosheet/public/v3/sheet_functions/string_functions.py:293]: Oh wait. I think I might be an idiot and it is possible for these functions to receive `None` as the parameters. `FIND(None, None)` does not error.

File: mitosheet/mitosheet/tests/public/v3/sheet_functions/string_functions/test_clean.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+from mitosheet.public.v3.sheet_functions.string_functions import LEN
+
+
+LEN_VALID_TESTS: Any = [
+    # Just constant tests
+    (['A'], 1),
+    (['ABC'], 3),
+    ([1.0], 3),
+    ([True], 4),
```
Comment (aarondr77) [inline mitosheet/mitosheet/tests/public/v3/sheet_functions/string_functions/test_clean.py:23]: We should add tests where there are None constants all of the parameters
